vertx-sql-client-template/src/main/java/io/vertx/sqlclient/template/generator/RowMapperGen.java

```diff
@@ -0,0 +1,215 @@
+            writer.print("    if (val != null) {\n");
+            writer.print("      for (" + rowType + " elt : (" + rowType + "[])val) {\n");
+            writer.print("        obj." + prop.getAdderMethod() + "(" + wrapExpr(prop.getType(), "elt") + ");\n");
+            writer.print("      }\n");
+            writer.print("    }\n");
+          }
+        }
+      });
+  }
+
+  private void bilto4(PrintWriter writer, String meth, PropertyInfo prop, String converter) {
+    String columnName = getMappingName(prop, Column.class.getName());
+    if (columnName != null) {
+      writer.print("    val = row." + meth + "(\"" + columnName + "\");\n");
+      writer.print("    if (val != null) {\n");
+      writer.print("      obj." + prop.getSetterMethod() + "(" + converter +  ");\n");
+      writer.print("    }\n");
+    }
+  }
+
+  private static String wrapExpr(TypeInfo type, String expr) {
+    DataObjectInfo dataObject = type.getDataObject();
+    if (dataObject != null) {
+      MapperInfo deserializer = dataObject.getDeserializer();
+      if (deserializer != null) {
+        if (deserializer.getKind() == MapperKind.SELF) {
+          return "new " + type.getName() + "(" + expr + ")";
+        } else {
+          return deserializer.getQualifiedName() + "." + String.join(".", deserializer
+            .getSelectors()) + "(" + expr + ")";
+        }
+      }
+      throw new UnsupportedOperationException();
+    } else {
+      return expr;
+    }
+  }
+
+  private static String rowType(TypeInfo type) {
+    DataObjectInfo dataObject = type.getDataObject();
+    if (dataObject != null) {
+      return rowType(dataObject.getJsonType());
+    }
+    return type.getName();
+  }
+
+  private static String getter(TypeInfo type) {
+    switch (type.getKind()) {
+      case PRIMITIVE:
+        PrimitiveTypeInfo pt = (PrimitiveTypeInfo) type;
+        return getter(pt.getBoxed());
+      case BOXED_PRIMITIVE:
+        return "get" + type.getSimpleName();
+      case STRING:
+        return "getString";
+      case JSON_OBJECT:
+        return "getJsonObject";
+      case JSON_ARRAY:
+        return "getJsonArray";
+    }
+    if (type instanceof ClassTypeInfo) {
+      ClassTypeInfo ct = (ClassTypeInfo) type;
+      switch (ct.getName()) {
+        case "java.time.LocalDateTime":
+          return "getLocalDateTime";
+        case "java.time.LocalDate":
+          return "getLocalDate";
+        case "java.time.LocalTime":
+          return "getLocalTime";
+        case "java.time.OffsetTime":
+          return "getOffsetTime";
+        case "java.time.OffsetDateTime":
+          return "getOffsetDateTime";
+        case "java.time.temporal.Temporal":
+          return "getTemporal";
+        case "java.util.UUID":
+          return "getUUID";
+        case "io.vertx.core.buffer.Buffer":
+          return "getBuffer";
+      }
+      DataObjectInfo dataObject = type.getDataObject();
```

Comment:
can we add tests for data object type like geometrics data object and numeric data object to check whether it works?

Reply:
yes, I'm not sure they work, we'll write the tests and figure out after :-)

Reply:
I added support and tests for such types

Reply:
Actually they are mapped to generic types with `row.get(Box.class, pos)`